Update K8up Documentation (#205)
vshn/k8up: docs/modules/ROOT/pages/references/config-reference.adoc

= Configuration Reference

> The Operator has two ways for configuration:

Reviewer: ```suggestion
The Operator can be configured in two ways:
```

. Per namespace backups. Optimal for shared clusters.
. Global settings with namespaced schedules. Optimal for private clusters.

== Environment Variables

> `BACKUP_ANNOTATION`:: the annotation to be used for filtering, default: `k8up.syn.tools/backup`

Reviewer: Filtering what?

> `BACKUP_ANNOTATION`:: the annotation to be used for filtering, default: `k8up.syn.tools/backup`

Reviewer: ```suggestion
`BACKUP_ANNOTATION`:: the annotation to be used for filtering, by default: `k8up.syn.tools/backup`
```

> `BACKUP_BACKUPCOMMANDANNOTATION`:: set the annotation name where the backup commands are stored, default `k8up.syn.tools/backupcommand`

Reviewer: ```suggestion
`BACKUP_BACKUPCOMMANDANNOTATION`:: set the annotation name that identify the backup commands on Pods, by default: `k8up.syn.tools/backupcommand`
```

> `BACKUP_CHECKSCHEDULE`:: the default check schedule, default: `0 0 * * 0`

Reviewer: ```suggestion
`BACKUP_CHECKSCHEDULE`:: the default check schedule, by default: `0 0 * * 0`
```

What is the check schedule for?

> `BACKUP_DATAPATH`:: where the PVCs should get mounted in the container, default `/data`

Reviewer: ```suggestion
`BACKUP_DATAPATH`:: to which path the PVCs should get mounted in the backup container, by default: `/data`
```

`BACKUP_FILEEXTENSIONANNOTATION`:: set the annotation name where the file extension is stored for backupcommands, default `k8up.syn.tools/file-extension`
`BACKUP_GLOBALACCESSKEYID`:: set the S3 access key id to be used globaly

> `BACKUP_GLOBALKEEPJOBS`:: set the count of jobs to keep globally

Reviewer: ```suggestion
`BACKUP_GLOBALKEEPJOBS`:: set the number of old backup jobs to keep globally
```

`BACKUP_GLOBALREPOPASSWORD`:: set the restic repository password to be used globaly

> `BACKUP_GLOBALRESTORES3ACCESKEYID`:: set the global restore S3 accessKeyID for restores

Reviewer: Do these perhaps default to the non-restore S3 pendant, i.e. `BACKUP_GLOBALACCESSKEYID` in this case?

Author: I have no idea :shrug:

`BACKUP_GLOBALRESTORES3BUCKET`:: set the global restore S3 bucket for restores
`BACKUP_GLOBALRESTORES3ENDPOINT`:: set the global restore S3 endpoint for the restores (needs the scheme [http/https]
`BACKUP_GLOBALRESTORES3SECRETACCESSKEY`:: set the global restore S3 SecretAccessKey for restores
`BACKUP_GLOBALS3BUCKET`:: set the S3 bucket to be used globally
`BACKUP_GLOBALS3ENDPOINT`:: set the S3 endpoint to be used globally
`BACKUP_GLOBALSECRETACCESSKEY`:: set the S3 secret access key to be used globaly
`BACKUP_GLOBALSTATSURL`:: set the URL of wrestic to post additional metrics gloablly, default `""`
`BACKUP_IMAGE`:: URL of the restic image, default: `172.30.1.1:5000/myproject/restic`
`BACKUP_JOBNAME`:: names for the backup job objects in OpenShift, default: `backupjob`

> `BACKUP_METRICBIND`:: set the bind address for the prometheus endpoint, default: `:8080`

Reviewer 2: I would love to see these environment variables being generated from code... this is a pain to maintain. Dunno how's it gonna look in K8up v2 though.
```suggestion
`BACKUP_METRICS_BINDADDRESS`:: set the bind address for the prometheus endpoint, default: `:8080`
```
 Also, some new variables are missing: 

- GlobalConcurrentArchiveJobsLimit
- GlobalConcurrentBackupJobsLimit
- GlobalConcurrentCheckJobsLimit
- GlobalConcurrentPruneJobsLimit
- GlobalConcurrentRestoreJobsLimit
- GlobalCPUResourceRequest
- GlobalCPUResourceLimit
- GlobalMemoryResourceRequest
- GlobalMemoryResourceLimit

`BACKUP_PODEXECACCOUNTNAME`:: set the service account name that should be used for the pod command execution, default: `pod-executor`
`BACKUP_PODEXECROLENAME`:: set the rolename that should be used for pod command execution, default `pod-executor`
`BACKUP_PODFILTER`:: the filter used to find the backup pods, default: `backupPod=true`
`BACKUP_PODNAME`:: names for the backup pod objects in OpenShift, default: `backupjob-pod`
`BACKUP_PROMURL`:: set the operator wide default prometheus push gateway, default `\http://127.0.0.1/`
`BACKUP_RESTARTPOLICY`:: set the RestartPolicy for the backup jobs. According to the https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/[docs] this should be `OnFailure` for jobs that terminate, default: `OnFailure`

> You only need to adjust `BACKUP_IMAGE` everything else can be left default.

Reviewer: ```suggestion
You only need to adjust `BACKUP_IMAGE`, everything else can be left default.
```

When do I have to adjust this ? If it's that important, maybe put this sentence at the beginning of the list, rather than the end.

== Global Settings

> Each variable starting with `BACKUP_GLOBAL*` is used to configure a global default for all namespaces.

Reviewer: ```suggestion
Each variable starting with `BACKUP_GLOBAL*` can be used to declare a global default for all namespaces.
```

> For example if you configure the S3 bucket and credentials you won’t have to specify them in the schedule or backup CRDs.

Reviewer: ```suggestion
For example, if you configure the S3 bucket and credentials here, you won’t have to specify them in the Schedule or Backup resource definitions.
```

> NOTE: It's possible to overwrite the global settings. Simply set the specific configuration in the CRD and it will use that instead.

Reviewer: ```suggestion
NOTE: It is always possible to overwrite the global settings. Simply declare the specific setting in the relevant resource definition and it will be applied instead of the global default.
```
